### DISCUSSION TITLE
ModuleNotFoundError: No module named 'ultima_scraper_api.managers

### DISCUSSION THREAD

**Discussion by machinesean** [DIGITALCRIMINALS/OnlyFans#893] [Q&A]
I created the Jan 2023 migration folders (whatever that means), updated the script and ran it, and now I'm getting this error whenever I try to run it;

PS D:\drivers\OnlyFans-7.6.1> poetry run python start_us.py
Traceback (most recent call last):
  File "D:\drivers\OnlyFans-7.6.1\start_us.py", line 24, in <module>
    main_test.check_start_up()
  File "D:\drivers\OnlyFans-7.6.1\tests\main_test.py", line 21, in check_start_up
    from ultima_scraper_api.managers.storage_managers.filesystem_manager import (
ModuleNotFoundError: No module named 'ultima_scraper_api.managers'

TIA.

**Comment by toine512** [comment]
Hi.

Same with both updated old version and master branch. (Windows)
`Traceback (most recent call last):
  File "Z:\UltimaScraper\start_us.py", line 8, in <module>
    from ultima_scraper_api.helpers.main_helper import OptionsFormat
ModuleNotFoundError: No module named 'ultima_scraper_api'`

**Comment by Patrick7070** [comment]
Are you running the updater.py script first before you try running the start_us.py ?
I got this error if I tried to start the scraper before running the updater

**Comment by toine512** [comment]
It works better! But still fails: `ModuleNotFoundError: No module named 'ultima_scraper_renamer'`

**Comment by machinesean** [comment]
same. I've gotten to this;

>  File "D:\drivers\OnlyFans-7.6.1\start_us.py", line 24, in <module>
>     main_test.check_start_up()
>   File "D:\drivers\OnlyFans-7.6.1\tests\main_test.py", line 21, in check_start_up
>     from ultima_scraper_api.managers.storage_managers.filesystem_manager import (
> ModuleNotFoundError: No module named 'ultima_scraper_api.managers'

**Comment by SiNevesh** [comment]
> Hi.
> 
> Same with both updated old version and master branch. (Windows) `Traceback (most recent call last): File "Z:\UltimaScraper\start_us.py", line 8, in <module> from ultima_scraper_api.helpers.main_helper import OptionsFormat ModuleNotFoundError: No module named 'ultima_scraper_api'`

I had this same issue. I fixed it by updating poetry dependencies. `poetry update` in terminal from the downloader folder.

**Comment by machinesean** [comment]
tried that, but I get hung up here:

> PS D:\drivers\OnlyFans-7.6.1> poetry update
> Updating dependencies
> Resolving dependencies...

and it never goes beyond that.

**Comment by Patrick7070** [comment]
That bug was mentioned on another thread.
You have to uninstall and reinstall poetry using pip
pip uninstall poetry
pip install poetry

And then run the updater.py script

**Comment by pzrd** [comment]
I've followed this thread and done everything as told, but now I get this error after installing poetry, uninstalling poetry, then reinstalling it and running updater.py twice and letting it download all those files. I have python version 3.10.4

>Traceback (most recent call last):
>File "C:\Users\A\OnlyFans-7.6.1\start_us.py", line 8, in <module>
    from ultima_scraper_api.helpers.main_helper import OptionsFormat
 >File "C:\Users\A\OnlyFans-7.6.1\ultima_scraper_api\__init__.py", line 5, in <module>
    from ultima_scraper_api.apis.fansly.fansly import start as FYStart
 >File "C:\Users\A\OnlyFans-7.6.1\ultima_scraper_api\apis\fansly\fansly.py", line 3, in <module>
    from ultima_scraper_api.apis.api_streamliner import StreamlinedAPI
 >File "C:\Users\A\OnlyFans-7.6.1\ultima_scraper_api\apis\api_streamliner.py", line 6, in <module>
    from ultima_scraper_api.apis.dashboard_controller_api import DashboardControllerAPI
 >File "C:\Users\A\OnlyFans-7.6.1\ultima_scraper_api\apis\dashboard_controller_api.py", line 5, in <module>
    import websockets
>ModuleNotFoundError: No module named 'websockets'

I did find that in my python site packages folder (c:\users\a\appdata\local\programs\python\python310\lib\site-packages) a Poetry folder was named as "~oetry" somehow and was creating an "Ignoring invalid distribution -oetry" error, could that be the issue? Is there a mistype in the updater maybe? I  renamed the folder to "poetry" in the site-packages folder, will try again.

Edit: so after renaming that site-package folder to poetry and then running updater.py again, I'm back to 

>ModuleNotFoundError: No module named 'ultima_scraper_api'

**Comment by Patrick7070** [comment]
When you do a pip list command do you see the websockets package installed ?  
If not do a pip install websockets
rename folder back to what it was ie ~oetry

**Comment by toine512** [comment]
Running `poetry update` as mentioned by @SiNevesh solved my issue. I guess it should be part of setup instructions?
